common/kvstore/key.go

```diff
@@ -0,0 +1,49 @@
+package kvstore
+
+import "errors"
+
+// ErrInvalidKey is returned when a key cannot be interpreted as the requested type.
+var ErrInvalidKey = errors.New("invalid key")
+
+// Key represents a key in a TableStore. Each key is scoped to a specific table.
+type Key interface {
+	// AsString interprets the key as a string and returns it.
```

Comment:
These are a lot of types, can we start with what we know we need?

Reply:
I've removed all of the methods except for the ones that deal with bytes. The other methods were not strictly necessary, just syntactic sugar.